app/gui/view/src/project.rs

```diff
@@ -258,12 +269,23 @@ impl Model {
         }
     }
 
-    fn on_dom_shape_changed(&self, shape: &dom::shape::Shape) {
-        // Top buttons must always stay in top-left corner.
+    // Relayouts the top bar elements and returns the new width of the top bar.
+    fn relayout_top_bar(
+        &self,
+        scene_shape: &display::scene::Shape,
+        window_control_buttons_width: f32,
+        dashboard_button_width: f32,
+    ) -> f32 {
+        let top_left = Vector2(-scene_shape.width, scene_shape.height) / 2.0;
         if let Some(window_control_buttons) = &*self.window_control_buttons {
-            let pos = Vector2(-shape.width, shape.height) / 2.0;
-            window_control_buttons.set_xy(pos);
+            window_control_buttons.set_xy(top_left);
         }
+        let dashboard_button_offset = Vector2(window_control_buttons_width, 0.0);
+        let dashboard_button_origin = Vector2(dashboard_button_width, -TOP_BAR_HEIGHT) / 2.0;
+        let dashboard_button_pos = top_left + dashboard_button_offset + dashboard_button_origin;
+        self.dashboard_button.set_xy(dashboard_button_pos);
+        let top_bar_width = dashboard_button_offset.x + dashboard_button_width;
+        top_bar_width
     }
```

Comment:
I can see that it can get confusing. Part of the top bar is defined by the project view, part of it is defined by the graph editor view. This is talking about the part that's defined by the view of _this module_ (the project view). Ideas for another name that would help make that distinction?

Reply:
The part that's defined in the project view contain the window control buttons and this newly added button to go to the project dashboard. The part that's defined in the graph editor view contain the execution environment dropdown, the project name and the breadcrumbs.